m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -118,23 +122,31 @@
 
     private final Set<MetricTag> commonTags;
 
-    // Holds metrics to be flushed
-    private final BlockingQueue<SizedMetric> metricQueue;
-    // The service used to execute metric flushes
-    private final ExecutorService executor;
+    // TODO replace w/ an evicting queue
```

Comment:
Another option here is to drop incoming metrics when the queue is full - this more closely matches the Go version of tally. However, we can also argue that recent metrics are more important than old metrics.

Reply:
Looks like we're now doing that, which is good--I see a change to use queue.offer instead of `queue.put`.